src/libstd/ffi/c_str.rs

```diff
@@ -844,6 +950,21 @@ impl CStr {
     ///
     /// [`Cow`]: ../borrow/enum.Cow.html
     /// [`str`]: ../primitive.str.html
+    ///
+    /// # Examples
+    ///
+    /// Calling `to_string_lossy` on a `CStr` with valid unicode:
+    ///
+    /// ```
+    /// use std::borrow::Cow;
+    /// use std::ffi::CStr;
+    ///
+    /// let c_str = CStr::from_bytes_with_nul(b"foo\0").unwrap();
+    /// assert_eq!(c_str.to_string_lossy(), Cow::Borrowed("foo"));
+    /// ```
+    ///
+    /// Had `c_str` contained invalid unicode, the `to_string_lossy` call might
+    /// have returned `Cow::Borrowed("fo�")`.
```

Comment:
> I figured it'd be undefined behavior to create a string w/ invalid unicode

`String`s and `str`s have to contain valid UTF-8 but `CString`s and `Cstr`s are just sequences of any none null bytes. The following example would be fine:

```rust
use std::borrow::Cow;
use std::ffi::CStr;

let c_str = CStr::from_bytes_with_nul(b"foo\xc3\0").unwrap();
assert_eq!(c_str.to_string_lossy(), "foo�");
```

It would also be more correct to say "(in)valid UTF-8" rather than "(in)valid unicode" in this case.